lib/postgrex/query.ex

```diff
@@ -22,10 +23,13 @@ defmodule Postgrex.Query do
     result_formats: [:binary | :text] | nil,
     decoders:       [Postgrex.Types.oid] | [(binary -> term)] | nil,
     types:          Postgrex.TypeServer.table | nil,
-    null:           atom}
+    null:           atom,
+    max_rows:       integer,
+    state:          :suspended | :done | nil
```

Comment:
Queries are reusable and including `state` breaks this guarantee. To maintain this we would have to store the portal state elsewhere, likely in the protocol state. This should mean that we don't need to return the query in the result and can reuse the query returned from `prepare/4`.